iceoryx2/src/service/static_config/message_type_details.rs

```diff
@@ -266,4 +266,130 @@ mod tests {
         let sut = unsafe { *payload_ptr };
         assert_that!(sut, eq demo.payload);
     }
+
+    #[test]
+    fn test_payload_layout() {
+        let details = MessageTypeDetails::from::<i64, i64, i64>(TypeVariant::FixedSize);
+        let sut = details.payload_layout(0);
+        assert_that!(sut.size(), eq 0);
+        let sut = details.payload_layout(5);
+        assert_that!(sut.size(), eq 40);
+
+        #[repr(C)]
+        struct Demo {
+            _b: bool,
+            _i16: i16,
+            _i64: i64,
+        }
+
+        let details = MessageTypeDetails::from::<i64, i64, Demo>(TypeVariant::FixedSize);
+        let sut = details.payload_layout(1);
+        #[cfg(target_pointer_width = "32")]
+        let expected = 12;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 16;
+        assert_that!(sut.size(), eq expected);
+
+        #[cfg(target_pointer_width = "32")]
+        let expected = 36;
+        #[cfg(target_pointer_width = "64")]
+        let expected = 48;
+        let sut = details.payload_layout(3);
+        assert_that!(sut.size(), eq expected);
+    }
+
+    #[test]
+    // test_sample_layout tests the sample layout for combinations of different types.
+    fn test_sample_layout() {
```

Comment:
I haven't found a better way to test it, so I calculated them in mind and test the layout for different architecture. Do you think it's fine or better idea?

Reply:
This is fine. Sometimes we have to go this route...